tv_apps/smart-system/js/splash_screen.js

```diff
@@ -56,5 +63,45 @@
     this.element.classList.add('hidden');
   };
 
+  SplashScreen.prototype.getIconUrl = function ss_getIconUrl(app) {
+    if (app.manifestURL) {
+      return this.getIconFromManifestURL(app.manifestURL);
+    } else if (app.features.iconUrl) {
+      return app.features.iconUrl;
+    }
+  };
+
+  SplashScreen.prototype.getIconFromManifestURL = function(manifestURL) {
+    var app = applications.getByManifestURL(manifestURL);
+    var icons = app.manifest.icons;
+
+    var maxIconSize = Object.keys(icons).reduce(function(prev, current) {
```

Comment:
There's a trick to get the largest number from an array: `Math.max(... Object.keys(icons))`, which is equal to `Math.max.apply(null, Object.keys(icons))`. :p

Again, it's up to you.